sql/src/main/java/io/crate/analyze/InsertAnalyzer.java

```diff
@@ -139,12 +143,53 @@ public AnalyzedInsertStatement analyze(Insert<?> insert, ParamTypeHints typeHint
         final boolean ignoreDuplicateKeys =
             insert.duplicateKeyContext().getType() == Insert.DuplicateKeyContext.Type.ON_CONFLICT_DO_NOTHING;
 
+        List<Symbol> returnValues = List.of();
+        List<ColumnIdent> outputNames = List.of();
+
+        if (!insert.returningClause().isEmpty()) {
+            var stmtCtx = new StatementAnalysisContext(typeHints, Operation.INSERT, txnCtx);
+            var relCtx = stmtCtx.startRelation();
+            var relation = relationAnalyzer.analyze(insert.table(), stmtCtx);
```

Comment:
```suggestion
            DocTableRelation targetTableRelation = (DocTableRelation) relationAnalyzer.analyze(insert.table(), stmtCtx);
```

Further below is currently a cast to `FieldResolver` that looks unsafe. 
My preference would be to put it here instead. Here the relation between `insert.table() -> DocTableRelation` is more visible. Could even add a comment to clarify that relationship.

Reply:
Or even better: 

```suggestion
            var relation = new DocTableRelation(tableInfo);
```